### PR TITLE
adding docs for engsys checks

### PR DESCRIPTION
**Pull request by seankane-msft**

### CODE REVIEW

File: doc/dev/engsys.md [hunk 1]
```diff
@@ -0,0 +1,73 @@
+# Engineering System Checks
```
Comment (scbedd) [inline doc/dev/engsys.md:1]: @seankane-msft  a ton of this is documented [here ](https://github.com/Azure/azure-sdk-for-python/blob/main/doc/eng_sys_checks.md)

We probably want to link to this doc from here.

Comment (seankane-msft) [reply doc/dev/engsys.md:1]: Didn't even see this, I'll move this to expand on the existing docs. Thanks for pointing it out scott

Comment (scbedd) [reply doc/dev/engsys.md:1]: When I read this I actually notice that a lot of the little "intro" details that are easily available here are not present in the other doc, so its not like this is pointless 👍 .

File: doc/dev/engsys.md [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+* [mypy](#mypy)
+* [black](#black)
+* [autorest automation](#autorest-automation)
```
Comment (kristapratico) [inline doc/dev/engsys.md:5]: Should we also add opt-in to running samples in live CI?

Comment (seankane-msft) [reply doc/dev/engsys.md]: Added

File: doc/dev/README.md [hunk 3]
```diff
@@ -3,8 +3,9 @@
 This folder contains documentation for developers of SDK: internal teams at Microsoft, or advanced contributors.
 
 Overview of the documents:
-- [dev setup](https://github.com/Azure/azure-sdk-for-python/blob/master/doc/dev/dev_setup.md) : How to create a development environment for this repo
-- [release](https://github.com/Azure/azure-sdk-for-python/blob/master/doc/dev/release.md) : How to release a package when ready
-- [packaging.md](https://github.com/Azure/azure-sdk-for-python/blob/master/doc/dev/packaging.md) : How to organize packaging information for packages under `azure`
+- [dev setup](https://github.com/Azure/azure-sdk-for-python/blob/main/doc/dev/dev_setup.md) : How to create a development environment for this repo
+- [release](https://github.com/Azure/azure-sdk-for-python/blob/main/doc/dev/release.md) : How to release a package when ready
+- [packaging.md](https://github.com/Azure/azure-sdk-for-python/blob/main/doc/dev/packaging.md) : How to organize packaging information for packages under `azure`
+- [engsys.md](https://github.com/Azure/azure-sdk-for-python/blob/main/doc/eng_sys_chekcs.md): Options for PR validation checks, how to opt-in, and how to run them locally
```
Comment (scbedd) [inline doc/dev/README.md:9]:
```suggestion
- [engsys.md](https://github.com/Azure/azure-sdk-for-python/blob/main/doc/eng_sys_checks.md): Options for PR validation checks, how to opt-in, and how to run them locally
```

File: doc/eng_sys_checks.md [hunk 4]
```diff
@@ -119,103 +159,143 @@ Regression also executes:
...
-3.  Install currently being developed version of package  we are testing regression for. E.g. azure-core 
-4.  Checkout the release tag of dependent package from github 
-5.  Install latest/oldest version of dependent package. For e.g. azure-identity 
-6.  Run test cases within dependent package from checked out branch.  
- 
- 
-Steps to manually run regression test locally: 
-1.  Run below command from your git code repo to generate the wheel of package being developed. Currently we have restricted to have prebuilt wheel. 
+
+Our regression framework automatically finds any such package that is added as required package so this list is not hardcoded.
```
Comment (scbedd) [inline doc/eng_sys_checks.md:225]:
```suggestion
Our regression framework automatically finds any such package that is added as required package, so this list is not hardcoded.
```

**Review by scbedd — APPROVED** [review]
Thanks @seankane-msft !

**Review by scbedd — APPROVED** [review]

**Review by kristapratico — APPROVED** [review]

**Pull request merged**